torch/_dynamo/optimizations/backends.py

```diff
@@ -785,6 +785,44 @@ def ltc_model(*inputs):
     return ltc_model
 
 
+@functools.lru_cache(None)
+def _init_torchxla():
+    global xm
+    try:
+        import torch_xla.core.xla_model as xm
+    except ModuleNotFoundError as e:
+        print(f"torchxla backend fails. Can not import {e.name}")
+        raise
+
+
+@create_backend
+def torchxla_trivial(subgraph):
+    _init_torchxla()
+
+    xla_dev = xm.xla_device()
+
+    xla_model = copy.deepcopy(subgraph.model).to(device=xla_dev)
+
+    def xla_model_wrapper(*inputs):
+        orig_device = inputs[0].device if len(inputs) > 0 else "cpu"
+        xla_inputs = tuple(inp.to(device=xla_dev) for inp in inputs)
```

Comment:
So basically you mean wrap the `model_iter_fn` once (to handle the cross device movements) so we don't need to do it in any other places?

I actually tried this idea since I agree that makes the code cleaner. But the problem I encountered is, the wrapped `model_iter_fn` fail to be compiled by dynamo. I guess there is some incompatibility if we apply XLA and dynamo (graph capture part rather than the backend) together.

Reply:
Since you are targeting on training later on, there is actually a head-up: random functions, e.g. torch.aten.ops.dropout, will require an extra seed input in torch-xla lowering. This seed, unfortunately, is not a PyTorch tensor input in the fx Graph, so there is no way to pass it with the `inp.to(device=xla)`. PyTorch pretends dropout are stateless and functional while it is really not.

This issue also applies to other non-Tensor scalars, but that is more of an issue for dynamo.